bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestWorkItem.java

```diff
@@ -30,26 +32,42 @@
 abstract class PullRequestWorkItem implements WorkItem {
     final Consumer<RuntimeException> errorHandler;
     final PullRequestBot bot;
-    final PullRequest pr;
+    final String prId;
+    PullRequest pr;
 
-    PullRequestWorkItem(PullRequestBot bot, PullRequest pr, Consumer<RuntimeException> errorHandler) {
+    PullRequestWorkItem(PullRequestBot bot, String prId, Consumer<RuntimeException> errorHandler) {
         this.bot = bot;
-        this.pr = pr;
+        this.prId = prId;
         this.errorHandler = errorHandler;
     }
 
     @Override
     public final boolean concurrentWith(WorkItem other) {
-        if (!(other instanceof PullRequestWorkItem)) {
+        if (!(other instanceof PullRequestWorkItem otherItem)) {
             return true;
         }
-        PullRequestWorkItem otherItem = (PullRequestWorkItem)other;
-        if (!pr.isSame(otherItem.pr)) {
+        if (!(prId.equals(otherItem.prId) && bot.repo().isSame(otherItem.bot.repo()))) {
```

Comment:
This means that two work requests can run concurrently if either the `prId` is different or the `bot repo` is different. I presume this is because the `prId` is relative to the repo (which makes sense), right?

Reply:
Correct. The prId is the number we see in PR links, e.g. 1349 for this one. The implementation of PullRequest::isSame is equivalent to the new code. The problem was that I can no longer guarantee to have the PullRequest object available when `concurrentWith` is called (rather, it's guaranteed to not be there), so I had to basically copy the code, and get the HostedRepository from the bot instead of the pr. This condition is supposed to be exactly the same as it was pre patch.